src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -85,46 +70,61 @@ public static String createNewIndex(Pinecone pinecone, int dimension, String ind
                 .spec(createIndexRequestSpec);
         pinecone.createIndex(createIndexRequest);
 
+        if (waitUntilIndexIsReady) {
+            waitUntilIndexIsReady(pinecone, indexName);
+        }
+
         return indexName;
     }
 
     public static IndexModel waitUntilIndexIsReady(Pinecone pinecone, String indexName, Integer totalMsToWait) throws InterruptedException {
         IndexModel index = pinecone.describeIndex(indexName);
         int waitedTimeMs = 0;
-        int intervalMs = 1500;
+        int intervalMs = 2000;
 
         while (!index.getStatus().getReady()) {
             index = pinecone.describeIndex(indexName);
             if (waitedTimeMs >= totalMsToWait) {
-                logger.info("Index " + indexName + " not ready after " + waitedTimeMs + "ms");
+                logger.info("WARNING: Index " + indexName + " not ready after " + waitedTimeMs + "ms");
                 break;
             }
             if (index.getStatus().getReady()) {
                 logger.info("Index " + indexName + " is ready after " + waitedTimeMs + "ms");
+                // Wait one final time before we start connecting and operating on the index
+                Thread.sleep(10000);
                 break;
             }
             Thread.sleep(intervalMs);
+            logger.info("Waited " + waitedTimeMs + "ms for " + indexName + " to get ready");
             waitedTimeMs += intervalMs;
         }
         return index;
     }
 
     public static IndexModel waitUntilIndexIsReady(Pinecone pinecone, String indexName) throws InterruptedException {
-        return waitUntilIndexIsReady(pinecone, indexName, 120000);
+        return waitUntilIndexIsReady(pinecone, indexName, 200000);
     }
 
-    public static PineconeConnection createNewIndexAndConnect(Pinecone pinecone, String indexName, int dimension, IndexMetric metric, CreateIndexRequestSpec spec) throws InterruptedException, PineconeException {
-        String apiKey = System.getenv("PINECONE_API_KEY");
+    public static Pinecone createNewIndex(Pinecone pinecone, String indexName, int dimension, IndexMetric metric, CreateIndexRequestSpec spec, boolean waitUntilIndexIsReady) throws InterruptedException, PineconeException {
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest().name(indexName).dimension(dimension).metric(metric).spec(spec);
+        pinecone.createIndex(createIndexRequest);
+
+        if (waitUntilIndexIsReady) {
+            waitUntilIndexIsReady(pinecone, indexName);
+        }
+        return pinecone;
```

Comment:
New `waitUntilIndexIsReady` `boolean` plus we return the `Pinecone` client now instead of a connection, the consumer can use `indexName` to create a connection.